src/main/java/com/bootcamp/demo/service/BookingService.java

```diff
@@ -0,0 +1,166 @@
+            return booking;
+        }
+        return null;
+    }
+
+    /**
+     * Retrieves a Booking entity by its user's id.
+     */
+    @Override
+    public LinkedHashSet<Booking> getBookings(final String userId) throws ExecutionException, InterruptedException, IllegalArgumentException {
+
+        return db.collection(COLLECTION_PATH)
+                .whereEqualTo("accountId", userId).get().get()
+                .getDocuments()
+                .stream().map(d -> d.toObject(Booking.class))
+                .collect(Collectors.toSet())
+                .stream().sorted(Comparator.comparing(Booking::getStartDate))
+                .collect(Collectors.toCollection(LinkedHashSet::new));
+    }
+
+    /**
+     * Retrieves all entities of type Booking.
+     */
+    @Override
+    public LinkedHashSet<Booking> getAllBookings() throws ExecutionException, InterruptedException {
+        LinkedHashSet<Booking> allBookings;
+        return allBookings = db.collection(COLLECTION_PATH)
+                .get()
+                .get()
+                .getDocuments()
+                .stream().map(d -> d.toObject(Booking.class))
+                .collect(Collectors.toSet())
+                .stream().sorted(Comparator.comparing(Booking::getStartDate))
+                .collect(Collectors.toCollection(LinkedHashSet::new));
+    }
+
+    /**
+     * Creates a new document containing the given Booking type param.
+     */
+    @Override
+    public String createBooking(final Booking booking) throws ExecutionException, InterruptedException, IllegalArgumentException {
+        if (booking == null) throw new IllegalArgumentException("Parameter 'booking' cannot be null");
+        return db.collection(COLLECTION_PATH)
+                .document(booking.getId())
+                .set(booking)
+                .get()
+                .getUpdateTime()
+                .toString();
+    }
+
+    @Override
+    public String deleteBooking(final String bookingId) throws ExecutionException, InterruptedException {
+        if (bookingId == null) throw new IllegalArgumentException("Parameter 'bookingId' cannot be null");
+        return db.collection(COLLECTION_PATH)
+                .document(bookingId)
+                .delete()
+                .get()
+                .getUpdateTime()
+                .toString();
+    }
+
+    @Override
+    public String updateBooking(String bookingId, String endDate, PaymentStatus newStatus) throws ExecutionException, InterruptedException {
+        if (bookingId == null) throw new IllegalArgumentException("Parameter 'bookingId' cannot be null");
+        return db.collection(COLLECTION_PATH)
+                .document(bookingId)
+                .update("endDate", endDate, "paymentStatus", newStatus)
+                .get()
+                .getUpdateTime()
+                .toString();
+    }
+
+    public Scooter bookAScooter(final Double lat, final Double longitude) throws ExecutionException, InterruptedException {

```

Comment:
Here I think we need another parameter for searchRadius which is expressed in m and not be hardcoded, so that the user can send this parameter.